src/models/AbstractWidget.php

```diff
@@ -0,0 +1,126 @@
+<?php namespace Leean\Widgets\Models;
+
+use Leean\Acf\All;
+
+/**
+ * Class AbstractWidget.
+ *
+ * @package Leean\Widgets\Models
+ */
+class AbstractWidget extends \WP_Widget
+{
+	/**
+	 * The slug.
+	 *
+	 * @var bool|string
+	 */
+	private $_slug = '';
+
+	/**
+	 * AbstractWidget constructor.
+	 *
+	 * @param string $title       Widget title
+	 * @param string $description Widget description.
+	 * @param bool   $slug		  Widget slug (use class name if not supplied).
+	 */
+	public function __construct( $title, $description, $slug = false ) {
```

Comment:
Yes I never said it can't add any number of methods that are worked by themselves but it requires to have at least 1 method to be abstract inside of the class that's the whole point to be abstract. 

And yes you are creating a new instance of this class on this line. https://github.com/moxie-leean/wp-widgets/pull/1/files#diff-3b5822c12b21544dd1b546beca29c8adR30 since PHP allow to do it with `ReflectionClass`